### PR TITLE
Changed disk sync based on new disk structure

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
tendrl-spec-id: Tendrl/specifications#159

Signed-off-by: GowthamShanmugam <gshanmug@redhat.com>

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
@r0h4n @brainfunked @shtripat @nnDarshan @nthomas-redhat @anivargi please review

**Comment by GowthamShanmugam** [comment]
Tested using VM and my local machine also

**Comment by shtripat** [comment]
@anivargi @cloudbehl keep an eye on this as this would affect UI and API layer .

File: tendrl/commons/objects/definition/master.yaml [hunk 1]
```diff
@@ -245,9 +245,87 @@ namespace.tendrl:
           help: "partitions details"
           type: json
       enabled: true
-      list: nodes/$NodeContext.node_id/Disks
-      value: nodes/$NodeContext.node_id/Disks
+      list: nodes/$NodeContext.node_id/LocalStorage/Disks
+      value: nodes/$NodeContext.node_id/LocalStorage/Disks
       help: "Disk"
+    VirtualDisk:
```
Comment (shtripat) [inline tendrl/commons/objects/definition/master.yaml]: Where is this class used?

Comment (GowthamShanmugam) [inline tendrl/commons/objects/definition/master.yaml]: @shtripat https://github.com/Tendrl/node-agent/pull/475/files#diff-96b0d41f23f2212ef7e267185ed18667R194

**Comment by shtripat** [comment]
LGTM to me. No major comments.

**Comment by r0h4n** [comment]
Please link to spec issue using "tendrl-bug-id: Tendrl/specifications#id", will review once that is done

File: tendrl/commons/objects/disk/__init__.py [hunk 2]
```diff
@@ -36,10 +36,15 @@ def __init__(self, disk_id=None, hardware_id=None, disk_name=None,
...
-                                       self.disk_id.replace('-', '_')
-                                       )
+        if self.disk_id == self.disk_name:
+            self.value = self.value.format(
+                NS.node_context.node_id,
+                self.disk_id.replace('/', '_').replace("_", "", 1))
+        else:
+            self.value = self.value.format(
+                NS.node_context.node_id,
+                self.disk_id)
```
Comment (r0h4n) [inline tendrl/commons/objects/disk/__init__.py]: no replace required here?

File: tendrl/commons/objects/virtual_disk/__init__.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    def __init__(self, disk_id=None, hardware_id=None, disk_name=None,
+                 sysfs_id=None, sysfs_busid=None, sysfs_device_link=None,
+                 hardware_class=None, model=None, vendor=None, device=None,
+                 rmversion=None, serial_no=None, driver=None, driver_modules=None,
+                 device_files=None, device_number=None, bios_id=None,
+                 geo_bios_edd=None, geo_logical=None, size=None, size_bios_edd=None,
+                 geo_bios_legacy=None, config_status=None, partitions=None,
+                 *args, **kwargs):
+        super(VirtualDisk, self).__init__(*args, **kwargs)
+        self.disk_id = disk_id
```
Comment (r0h4n) [inline tendrl/commons/objects/virtual_disk/__init__.py]: cant you inherit these common attrs from the disk object?

File: tendrl/commons/objects/disk/__init__.py (hunk 2 above)
Comment (GowthamShanmugam) [inline tendrl/commons/objects/disk/__init__.py]: @r0h4n here no replace is required, here the key will come like vendor_device_serialNo

File: tendrl/commons/objects/virtual_disk/__init__.py (hunk 3 above)
Comment (GowthamShanmugam) [inline tendrl/commons/objects/virtual_disk/__init__.py]: @r0h4n ok

File: tendrl/commons/objects/virtual_disk/__init__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+                 *args, **kwargs):
+        super(VirtualDisk, self).__init__(
+            disk_id, hardware_id, disk_name, sysfs_id, sysfs_busid,
+            sysfs_device_link, hardware_class, model, vendor, device,
+            rmversion, serial_no, driver, driver_modules, device_files,
+            device_number, bios_id, geo_bios_edd, geo_logical, size,
+            size_bios_edd, geo_bios_legacy, config_status, partitions,
+            *args, **kwargs)
+        self.value = 'nodes/{0}/LocalStorage/Virtio/{1}'
+
```
Comment (GowthamShanmugam) [inline tendrl/commons/objects/virtual_disk/__init__.py]: parent render() is reused for VirtualDisk

**Comment by GowthamShanmugam** [comment]
@r0h4n fixed

**Comment by anivargi** [comment]
Looks good from the backend and API front, need @cloudbehl @gnehapk to ack all attributes are present.

**Pull request merged**